### PR TITLE
Update packages for wagtail, django

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: molo/core/admin.py [hunk 1]
```diff
@@ -126,8 +127,8 @@ class ArticleModelAdmin(WagtailModelAdmin, ArticleAdmin):
     list_filter = [
         StatusCategoryListFilter,
         SectionListFilter,
-        ('first_published_at', DateFilter),
-        ('latest_revision_created_at', DateFilter),
+        # ('first_published_at', DateFilter),
+        # ('latest_revision_created_at', DateFilter),
```
Comment (KaitCrawford) [inline molo/core/admin.py]: Why are you removing these?

File: molo/core/api/urls.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
-from django.conf.urls import url
+from django.conf.urls import re_path
 
+from wagtail.api.v2.views import PagesAPIViewSet
```
Comment (KaitCrawford) [inline molo/core/api/urls.py]: I don't see this import being used anywhere?

File: molo/core/templatetags/core_tags.py [hunk 3]
```diff
@@ -31,7 +33,7 @@ def get_language(site, locale):
     if not language:
         language = Languages.for_site(site).languages.filter(
             locale=locale).first()
-        cache.set(language_cache_key, language, None)
+        # cache.set(language_cache_key, language, None)
```
Comment (KaitCrawford) [inline molo/core/templatetags/core_tags.py]: Why are we not caching this anymore?

File: molo/core/tests/test_translations.py [hunk 4]
```diff
@@ -257,6 +256,8 @@ def test_that_only_translated_sections_show_with_tag_navigation(self):
         site_settings.enable_tag_navigation = True
         site_settings.show_only_translated_pages = True
         site_settings.save()
+        response = self.client.get('/locale/fr/')
+        response = self.client.get('/')
```
Comment (KaitCrawford) [inline molo/core/tests/test_translations.py:262]: Why are these lines needed here? before we've even created the french section

File: molo/core/tests/test_views.py [hunk 5]
```diff
@@ -1789,7 +1789,7 @@ def test_next_article_main_language(self):
 
     def test_next_article_not_main_language(self):
         # assumes articles loop
-        self.client.get('/locale/fr/')
+        response = self.client.get('/locale/fr/')
```
Comment (KaitCrawford) [inline molo/core/tests/test_views.py]: Why do you now need to assign this to a variable?

File: molo/core/views.py [hunk 6]
```diff
@@ -100,7 +100,6 @@ def search(request, results_per_page=10, load_more=False):
 
 
 def locale_set(request, locale):
-    request.session[LANGUAGE_SESSION_KEY] = locale
```
Comment (KaitCrawford) [inline molo/core/views.py:103]: Why are we not setting the session key anymore?

File: molo/crontab.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+0 22,10 * * * cd /app && /usr/local/bin/python rapidpro_to_bigquery.py >> /var/log/cron.log 2>&1
```
Comment (KaitCrawford) [inline molo/crontab.txt]: What is this for/about?

File: molo/profiles/migrations/0006_auto_20201208_0613.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        ('profiles', '0005_auto_20190723_1808'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='userprofile',
+            name='site',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, to='wagtailcore.site'),
+        ),
+    ]
```
Comment (KaitCrawford) [inline molo/profiles/migrations/0006_auto_20201208_0613.py:20]: Where is this migration coming from? I don't see any changes to the model below

File: molo/core/admin.py [hunk 9]
```diff
@@ -1,23 +1,24 @@
-from daterange_filter.filter import DateRangeFilter
+# from daterange_filter.filter import DateRangeFilter
```
Comment (KaitCrawford) [inline molo/core/admin.py]: Why are we removing these filters from the admin?

File: molo/core/middleware.py [hunk 10]
```diff
@@ -102,6 +103,16 @@ def process_request(self, request):
...
 
 
+class SetLangaugeCodeMiddleware(django.utils.deprecation.MiddlewareMixin):
+    """Sets the language code"""
+    def process_response(self, request, response):
+        if not 'locale' in request.path:
+            return response
+        locale_code = request.path.split("/")[2]
+        response.set_cookie('django_language', locale_code)
+        return response
```
Comment (KaitCrawford) [inline molo/core/middleware.py:112]: Why is this new middleware needed? where was this previously set?

File: molo/core/models.py [hunk 11]
```diff
@@ -107,14 +107,6 @@ class ArticleOrderingChoices(enum.Enum):
...
     PK_DESC = 5
 
-    labels = {
-        CMS_DEFAULT_SORTING: 'CMS Default Sorting',
-        FIRST_PUBLISHED_AT: 'First Published At',
-        FIRST_PUBLISHED_AT_DESC: 'First Published At Desc',
-        PK: 'Primary Key',
-        PK_DESC: 'Primary Key Desc',
-    }
-
```
Comment (KaitCrawford) [inline molo/core/models.py:117]: Don't we use these labels for the site settings?

File: molo/core/migrations/0021_auto_20201208_0615.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+        ),
+    ]
```
Comment (KaitCrawford) [inline molo/core/migrations/0021_auto_20201208_0615.py:31]: What caused these changes?

File: molo/profiles/admin.py [hunk 13]
```diff
@@ -5,12 +5,12 @@
...
 from django.contrib.auth.admin import UserAdmin
 from django.contrib.admin.sites import NotRegistered
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 from django.utils.timezone import localtime
 from django.conf import settings
 from django.db.models import Q
 
-from daterange_filter.filter import DateRangeFilter
+# from daterange_filter.filter import DateRangeFilter
```
Comment (KaitCrawford) [inline molo/profiles/admin.py]: Why are we removing this filter?

File: molo/requirements.txt [hunk 14]
```diff
@@ -0,0 +1,3 @@
+rapidpro-python==2.6.1
+NewlineJSON
+google-cloud-bigquery
```
Comment (KaitCrawford) [inline molo/requirements.txt]: Why do we need to install the bigquery and rapidpro libraries? what are we using them for?

**Comment by KaitCrawford** [comment]
When removing lines please delete them rather than commenting them out (eg the admin filters)

File: molo/core/middleware.py (hunk 10 above)
Comment (codiebeulaine) [reply molo/core/middleware.py:112]: previously it was in set in session, now django 3 changed their ways to check cookie not session

File: molo/core/migrations/0021_auto_20201208_0615.py (hunk 12 above)
Comment (codiebeulaine) [reply molo/core/migrations/0021_auto_20201208_0615.py:31]: That was from previous changes. When I started these migrations needed to be made

File: molo/core/tests/test_translations.py (hunk 4 above)
Comment (codiebeulaine) [reply molo/core/tests/test_translations.py:262]: to set the locale as french

File: molo/core/views.py (hunk 6 above)
Comment (codiebeulaine) [reply molo/core/views.py:103]: it now gets it from cookie

File: molo/profiles/migrations/0006_auto_20201208_0613.py (hunk 8 above)
Comment (codiebeulaine) [reply molo/profiles/migrations/0006_auto_20201208_0613.py:20]: was created on fresh instance

File: molo/profiles/admin.py (hunk 13 above)
Comment (codiebeulaine) [reply molo/profiles/admin.py]: not compatibale, looking at alternatives

File: molo/core/admin.py [hunk 15]
```diff
@@ -1,25 +1,21 @@
-from daterange_filter.filter import DateRangeFilter
+from rangefilter.filter import DateRangeFilter, DateTimeRangeFilter
```
Comment (KaitCrawford) [inline molo/core/admin.py]: I don't think we're using DateTimeRangeFilter are we?

File: molo/core/models.py (hunk 11 above)
Comment (codiebeulaine) [reply molo/core/models.py:117]: added in django built in enum support

Comment (codiebeulaine) [reply molo/core/models.py:117]: changed on top

File: molo/core/templatetags/core_tags.py [hunk 16]
```diff
@@ -27,11 +29,14 @@
...
-    if not language:
+    language_pk = cache.get(language_cache_key)
+    if not language_pk:
         language = Languages.for_site(site).languages.filter(
             locale=locale).first()
-        cache.set(language_cache_key, language, None)
+        if language:
+            cache.set(language_cache_key, language.pk, None, None)
+    else:
+        language = SiteLanguageRelation.objects.get(pk=language_pk)
```
Comment (KaitCrawford) [inline molo/core/templatetags/core_tags.py:39]: This would return a SiteLanguageRelation object, not a Language object. Are you sure that's what we want? Does the `Languages.for_site().languages...` call also return a SiteLanguageRelation object?

File: molo/profiles/admin.py (hunk 13 above)
Comment (KaitCrawford) [reply molo/profiles/admin.py]: You changed to the new filter in the core admin. why not here as well?
If we don't want a filter here then please remove this code rather than commenting it out

**Review by KaitCrawford — COMMENTED** [review]
Some minor questions/comments

File: molo/core/templatetags/core_tags.py (hunk 16 above)
Comment (codiebeulaine) [reply molo/core/templatetags/core_tags.py:39]: yep

File: molo/profiles/templates/rangefilter/date_filter.html [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+File: molo/core/templates/daterange_filter/filter.html
+
+This file is empty because:
+
+- The django-daterange-filter package is unmaintained and does not support
+  newer versions of Django admin
+- If this template does not exist Django raises an exception
+- Removing the package completely is hard because we use it with our own
+  custom templates in the Wagtail admin
```
Comment (KaitCrawford) [inline molo/profiles/templates/rangefilter/date_filter.html:11]: but you have removed the package? it's no longer part of the setup.py file?

Comment (codiebeulaine) [reply molo/profiles/templates/rangefilter/date_filter.html:11]: We changed the package. Still needs template

**Review by KaitCrawford — APPROVED** [review]
